packages/riverpod/lib/src/state_notifier_provider.dart

```diff
@@ -91,9 +91,37 @@ mixin _StateNotifierStateProviderStateMixin<T>
     exposedValue = value;
   }
 
+  @override
+  void exposedValueChanged(T newValue) => exposedValue = newValue;
```

Comment:
unnecessary override?

Reply:
Wait instead, shouldn't use it `addListener`?

Reply:
it is stateNotifierProvider.state. It only needs to listen during valueChanged becaus it will never change the StateNotifier via setState. Also , override is unnecessary, yes :) will be fixed soon